diff --git a/node_modules/hyperswarm/index.js b/node_modules/hyperswarm/index.js
index a242046..97ce336 100644
--- a/node_modules/hyperswarm/index.js
+++ b/node_modules/hyperswarm/index.js
@@ -104,9 +104,9 @@ module.exports = class Hyperswarm extends EventEmitter {
     this.on('update', this._handleUpdate)
   }
 
-  _maybeRelayConnection(force) {
+  _maybeRelayConnection(force, peerInfo) {
     if (!this.relayThrough) return null
-    return this.relayThrough(force, this)
+    return this.relayThrough(force, this, peerInfo)
   }
 
   _enqueue(peerInfo) {
@@ -207,7 +207,7 @@ module.exports = class Hyperswarm extends EventEmitter {
       return
     }
 
-    const relayThrough = this._maybeRelayConnection(peerInfo.forceRelaying)
+    const relayThrough = this._maybeRelayConnection(peerInfo.forceRelaying, peerInfo)
     const conn = this.dht.connect(peerInfo.publicKey, {
       relayAddresses: peerInfo.relayAddresses,
       keyPair: this.keyPair,
